src/middleware.ts

```diff
@@ -137,12 +143,24 @@ export async function authenticateRoute(request: NextRequest) {
   // if cookie expired, run refresh routine
   if (Date.now() > authObj.expiry) {
     const newAuthResponse = await refreshAuthToken(authObj.refreshToken);
-    let nextRes = NextResponse.next();
-    nextRes = setAuthCookie(newAuthResponse, nextRes);
-    return nextRes;
+    response = setAuthCookie(newAuthResponse, response);
+    return response;
   }
-  // cookie is not expired, go to page
-  return NextResponse.next();
+  // they're logged in already
+  response = setLastViewedOrg(request, response);
+  // go to page
+  return response;
+}
+
+export function setLastViewedOrg(request: NextRequest, response: NextResponse) {
+  const matches = request.nextUrl.pathname.match(
+    /orgs\/([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12})/
```

Comment:
[question] Regex intimidates me - does this match a GUID?

Reply:
It does! (Thanks, Stack Overflow). I need this GUID regex in a few places, so I tried to make it a variable, but I was having trouble inserting the variable into the larger regex pattern here. If you have any ideas there on how to make that more DRY, would love to hear them.